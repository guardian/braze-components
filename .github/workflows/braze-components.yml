name: Build braze-components

on: push

jobs:
  braze-components-build:
    name: braze-components-test-and-build
    runs-on: ubuntu-latest

    if: >-
      (github.actor != 'dependabot[bot]') &&
        (github.event.pull_request.head.repo.owner.login == 'guardian' ||
          github.event_name == 'push')

    # Required by actions-assume-aws-role
    permissions:
      id-token: write
      contents: read

    steps:
      - name: Checkout repo
        uses: actions/checkout@v2

      - name: Use Node.js
        uses: actions/setup-node@main

      - name: Install dependencies
        run: yarn install

      - name: Check typescript
        run: yarn tsc

      - name: Lint
        run: yarn lint

      - name: Run tests
        run: CI=true yarn test

      - name: Storybook build
        run: yarn build-storybook

      - name: CDK
        working-directory: cdk
        run: ./script/ci 

      - name: Upload
        run: |
          export LAST_TEAMCITY_BUILD=1000
          export GITHUB_RUN_NUMBER=$(( $GITHUB_RUN_NUMBER + $LAST_TEAMCITY_BUILD ))
          yarn upload-artifact


# FROM THE TYPESCRIPT GHA
# jobs:
#     typescript:
#         runs-on: ubuntu-latest
#         steps:
#             - uses: actions/checkout@v2.4.0
#             - uses: guardian/actions-setup-node@main

#             # Cache npm dependencies using https://github.com/bahmutov/npm-install
#  >>> THIS NEXT LINE???
#             - uses: bahmutov/npm-install@v1

#             - name: Install
#               run: yarn

#             - name: Check typescript
#               run: yarn tsc
