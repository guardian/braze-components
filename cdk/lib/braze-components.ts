import { join } from 'path';
import type { CfnDistribution } from '@aws-cdk/aws-cloudfront';
import { CloudFrontWebDistribution, OriginAccessIdentity } from '@aws-cdk/aws-cloudfront';
import { Bucket } from '@aws-cdk/aws-s3';
import { CfnInclude } from '@aws-cdk/cloudformation-include';
import type { App } from '@aws-cdk/core';
import { Duration } from '@aws-cdk/core';
import type { GuStackProps } from '@guardian/cdk/lib/constructs/core';
import { GuStack, GuStageParameter } from '@guardian/cdk/lib/constructs/core';
import { GuCname } from '@guardian/cdk/lib/constructs/dns';

export class BrazeComponents extends GuStack {
    constructor(scope: App, id: string, props: GuStackProps) {
        super(scope, id, props);
        const yamlTemplateFilePath = join(__dirname, '../..', 'cloudformation.yaml');
        const template = new CfnInclude(this, 'YamlTemplate', {
            templateFile: yamlTemplateFilePath,
            parameters: {
                Stage: GuStageParameter.getInstance(this),
            },
        });

        const oldCloudfrontDist = template.getResource('CDN') as CfnDistribution;

        new GuCname(this, 'DNS entry', {
            app: 'braze-components',
            domainNameProps: {
                CODE: { domainName: 'braze-components.code.dev-gutools.co.uk' },
                PROD: { domainName: 'braze-components.gutools.co.uk' },
            },
            resourceRecord: oldCloudfrontDist.getAtt('DomainName').toString(),
            ttl: Duration.hours(1),
        });

        const bucketNameFromStaticCloudformationStack = 'braze-components-storybook';
        const sourceBucket = Bucket.fromBucketName(
            this,
            'braze-components-bucket',
            bucketNameFromStaticCloudformationStack,
        );

        const originAccessIdFromStaticCloudformationStack = 'E3EA9DC41190PP';
        const originAccessIdentity = OriginAccessIdentity.fromOriginAccessIdentityName(
            this,
            'braze-components-origin-access-identity',
            originAccessIdFromStaticCloudformationStack,
        );

        new CloudFrontWebDistribution(this, 'braze-components-cloudfront', {
            originConfigs: [
                {
                    s3OriginSource: {
                        s3BucketSource: sourceBucket,
                        originAccessIdentity,
                        originPath: `/${this.stage}/braze-components-storybook-static`,
                    },
                    behaviors: [{ isDefaultBehavior: true }],
                },
            ],
        });
    }
}
