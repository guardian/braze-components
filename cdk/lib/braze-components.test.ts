import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { BrazeComponents } from './braze-components';

describe('The BrazeComponents stack', () => {
    it('matches the snapshot', () => {
        const app = new App();
        const stack = new BrazeComponents(app, 'BrazeComponents', {
            stack: 'targeting',
            stage: 'TEST',
            domainName: 'braze-components.test.dev-gutools.co.uk',
            tlsCertId: 'xxx-xxx-xxx',
        });
        const template = Template.fromStack(stack);
        expect(template.toJSON()).toMatchSnapshot();
    });
});
