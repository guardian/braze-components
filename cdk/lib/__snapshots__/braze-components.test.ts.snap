// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The BrazeComponents stack matches the snapshot 1`] = `
Object {
  "Metadata": Object {
    "gu:cdk:constructs": Array [
      "GuCname",
    ],
    "gu:cdk:version": "TEST",
  },
  "Resources": Object {
    "CDN": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "Aliases": Array [
            "braze-components.test.dev-gutools.co.uk",
          ],
          "DefaultCacheBehavior": Object {
            "AllowedMethods": Array [
              "GET",
              "HEAD",
            ],
            "CachedMethods": Array [
              "GET",
              "HEAD",
            ],
            "Compress": true,
            "ForwardedValues": Object {
              "Cookies": Object {
                "Forward": "none",
              },
              "QueryString": false,
            },
            "TargetOriginId": "origin1",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Origins": Array [
            Object {
              "ConnectionAttempts": 3,
              "ConnectionTimeout": 10,
              "DomainName": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "braze-components-storybook.s3.",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ".",
                    Object {
                      "Ref": "AWS::URLSuffix",
                    },
                  ],
                ],
              },
              "Id": "origin1",
              "OriginPath": "/TEST/braze-components-storybook-static",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": "origin-access-identity/cloudfront/E3EA9DC41190PP",
              },
            },
          ],
          "PriceClass": "PriceClass_100",
          "ViewerCertificate": Object {
            "AcmCertificateArn": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "arn:aws:acm:us-east-1:",
                  Object {
                    "Ref": "AWS::AccountId",
                  },
                  ":certificate/xxx-xxx-xxx",
                ],
              ],
            },
            "SslSupportMethod": "sni-only",
          },
        },
        "Tags": Array [
          Object {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          Object {
            "Key": "gu:repo",
            "Value": "guardian/braze-components",
          },
          Object {
            "Key": "Stack",
            "Value": "targeting",
          },
          Object {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "DNSentry": Object {
      "Properties": Object {
        "Name": "braze-components.test.dev-gutools.co.uk",
        "RecordType": "CNAME",
        "ResourceRecords": Array [
          Object {
            "Fn::GetAtt": Array [
              "CDN",
              "DomainName",
            ],
          },
        ],
        "Stage": "TEST",
        "TTL": 3600,
      },
      "Type": "Guardian::DNS::RecordSet",
    },
  },
}
`;
